File: app/src/main/java/com/capstone/nik/mixology/Network/CocktailService.java
package com.capstone.nik.mixology.Network;

import com.capstone.nik.mixology.Model.Cocktail;
import com.capstone.nik.mixology.Network.remoteModel.Cocktails;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by nik on 1/28/2018.
 */

public interface CocktailService {

    // Has Alcohol Filter
    @GET("filter.php")
    Call<Cocktails> getAlcoholFilter(@Query("a") String filter);

    // Glass Type Filter
    @GET("filter.php")
    Call<Cocktails> getGlassFilter(@Query("g") String filter);

    // Ingredient Filter
    @GET("filter.php")
    Call<Cocktails> getIngredientFilter(@Query("i") String filter);

    // Drink Category (Type) Filter
    @GET("filter.php")
    Call<Cocktails> getDrinkTypeFilter(@Query("c") String filter);

    // Search
    @GET("search.php")
    Call<Cocktails> getSearchResults(@Query("s") String search);

    //Randomixer
    @GET("random.php")
    Call<Cocktails> getRandomixer();

    // Search by id
    @GET("lookup.php")
    Call<Cocktails> getDrinkById(@Query("i") String id);
}

File: app/src/main/java/com/capstone/nik/mixology/Network/CocktailService.kt
package com.capstone.nik.mixology.Network

import com.capstone.nik.mixology.Network.remoteModel.CatalogListResponse
import com.capstone.nik.mixology.Network.remoteModel.CocktailDbResponse
import retrofit2.http.GET
import retrofit2.http.Query

interface CocktailService {

    @GET("filter.php")
    suspend fun getAlcoholFilter(@Query("a") filter: String): CocktailDbResponse

    @GET("filter.php")
    suspend fun getGlassFilter(@Query("g") filter: String): CocktailDbResponse

    @GET("filter.php")
    suspend fun getIngredientFilter(@Query("i") filter: String): CocktailDbResponse

    @GET("filter.php")
    suspend fun getDrinkTypeFilter(@Query("c") filter: String): CocktailDbResponse

    @GET("search.php")
    suspend fun getSearchResults(@Query("s") search: String): CocktailDbResponse

    @GET("random.php")
    suspend fun getRandomixer(): CocktailDbResponse

    @GET("lookup.php")
    suspend fun getDrinkById(@Query("i") id: String): CocktailDbResponse

    @GET("search.php")
    suspend fun getSearchByLetter(@Query("f") letter: String): CocktailDbResponse

    @GET("list.php")
    suspend fun listCategories(@Query("c") list: String = "list"): CatalogListResponse

    @GET("list.php")
    suspend fun listGlasses(@Query("g") list: String = "list"): CatalogListResponse

    @GET("list.php")
    suspend fun listIngredients(@Query("i") list: String = "list"): CatalogListResponse

    @GET("list.php")
    suspend fun listAlcoholic(@Query("a") list: String = "list"): CatalogListResponse
}
